docker-compose.debug.yml

```diff
@@ -0,0 +1,42 @@
+#
+# docker-compose file configured for debugging the flask app.
+# Should be used in conjunction with a vs code launch configuration
+# for attaching a debugger to the flask process (see the
+# `Python: Attach` config in launch.json). This allows the addition
+# of breakpoints to the flask app's python code from vs code.
+#
+version: "3.7"
+services:
+  flask-dev:
+    container_name: metrics-flask-dev
+    build:
+      dockerfile: Dockerfile
+      target: flask-dev
+      context: .
+    command: ["sh", "-c", "pip install debugpy -t /tmp && python /tmp/debugpy --wait-for-client --listen 0.0.0.0:6789 -m flask run --no-debugger --no-reload --host 0.0.0.0 --port 5000"]
```

Comment:
Both this and the collector is starting to build up a lot of docker-compose files (I'm guilty of adding to this on the collector side), I wonder if since this compose file seems to only differ in overriding the Dockerfile `CMD` step for the `metrics-flask-dev` image, if we can make the build command (seen here https://github.com/codeforpdx/opentransit-metrics/blob/75ccd2bdcfa0581a6f50038ce84612b08d1795c5/Dockerfile#L21) an `ARG` that can be passed in from the VSCode Launch config.

I'd need to mess around with this a bit. Potentially a future enhancement.

Reply:
Yeah, I know what you mean about having a lot of docker-compose files! :) I'm not sure how much config is possible to pass in to a docker-compose invocation from vscode. It's definitely a little more straightforward to customize starting up a single docker image from vscode. Or at least most of the vs code debug docs for docker compose focused on having some sort of debug file like this.